docs/csharp/reference-semantics-with-value-types.md

```diff
@@ -88,17 +88,39 @@ temporary variable created as part of the method call.
 There are several ways in which the compiler ensures that the read-only
 nature of an `in` argument is enforced.  First of all, the called method
 can't directly assign to an `in` parameter. It can't directly assign
-to any field of an `in` parameter. In addition, you cannot pass
-an `in` parameter to any method demanding the `ref` or `out` modifier.
-The compiler enforces that the `in` argument is a readonly variable. You
-can call any instance method that uses pass-by-value semantics. In
+to any field of an `in` parameter when that value is a `struct` type. In addition, you cannot pass
+an `in` parameter to any method using the `ref` or `out` modifier.
+These rules apply to any field of an `in` parameter provided the
+field is a `struct` type and the parameter is also a `struct` type. In fact, these rules
+apply for multiple layers of member access provided the types at all levels
+of member access are `structs`.
+The compiler enforces the `in` argument and its `struct` members are readonly variables.
```

Comment:
> The compiler enforces the `in` argument and its `struct` members are readonly variables.

This implies that a `struct` field of a `class` `in` argument should be readonly, but that's not true.

I think the explanation before that is correct, it's just the summary in the last sentence that is problematic.